router/usecase/pools/routable_cw_alloy_transmuter_pool.go

```diff
@@ -0,0 +1,177 @@
+func (r *routableAlloyTransmuterPoolImpl) SetTokenOutDenom(tokenOutDenom string) {
+	r.TokenOutDenom = tokenOutDenom
+}
+
+// CalcSpotPrice implements sqsdomain.RoutablePool.
+func (r *routableAlloyTransmuterPoolImpl) CalcSpotPrice(ctx context.Context, baseDenom string, quoteDenom string) (osmomath.BigDec, error) {
+	return r.CalcTokenOutAmt(sdk.Coin{Denom: baseDenom, Amount: osmomath.OneInt()}, quoteDenom)
+}
+
+// IsGeneralizedCosmWasmPool implements sqsdomain.RoutablePool.
+func (*routableAlloyTransmuterPoolImpl) IsGeneralizedCosmWasmPool() bool {
+	return false
+}
+
+// GetCodeID implements sqsdomain.RoutablePool.
+func (r *routableAlloyTransmuterPoolImpl) GetCodeID() uint64 {
+	return r.ChainPool.CodeId
+}
+
+// FindNormalizationFactors finds the normalization factors for the given token in and token out denoms.
+// It is required for calculating token out & spot price.
+// For more information about normalization factor, please refer to [transmuter documentation](https://github.com/osmosis-labs/transmuter/tree/v3.0.0?tab=readme-ov-file#normalization-factors).
+func (r *routableAlloyTransmuterPoolImpl) FindNormalizationFactors(tokenInDenom, tokenOutDenom string) (osmomath.Int, osmomath.Int, error) {
+	tokenInNormalizationFactor := osmomath.Int{}
+	tokenOutNormalizationFactor := osmomath.Int{}
+
+	for _, config := range r.AlloyTransmuterData.AssetConfigs {
+		if config.Denom == tokenInDenom {
+			tokenInNormalizationFactor = config.NormalizationFactor
+		}
+
+		if config.Denom == tokenOutDenom {
+			tokenOutNormalizationFactor = config.NormalizationFactor
+		}
+
+		if !tokenInNormalizationFactor.IsNil() && !tokenOutNormalizationFactor.IsNil() {
+			break
+		}
+	}
+
+	if tokenInNormalizationFactor.IsNil() {
+		return tokenInNormalizationFactor, tokenOutNormalizationFactor, domain.MissingNormalizationFactorError{Denom: tokenInDenom, PoolId: r.GetId()}
+	}
+
+	if tokenOutNormalizationFactor.IsNil() {
+		return tokenInNormalizationFactor, tokenOutNormalizationFactor, domain.MissingNormalizationFactorError{Denom: tokenOutDenom, PoolId: r.GetId()}
+	}
+
+	return tokenInNormalizationFactor, tokenOutNormalizationFactor, nil
+}
+
+// Calculate the token out amount based on the normalization factors:
+//
+// token_out_amt / token_out_norm_factor = token_in_amt / token_in_norm_factor
+// token_out_amt = token_in_amt * token_out_norm_factor / token_in_norm_factor
+func (r *routableAlloyTransmuterPoolImpl) CalcTokenOutAmt(tokenIn sdk.Coin, tokenOutDenom string) (osmomath.BigDec, error) {
+	tokenInNormFactor, tokenOutNormFactor, err := r.FindNormalizationFactors(tokenIn.Denom, tokenOutDenom)
+	if err != nil {
+		return osmomath.ZeroBigDec(), err
+	}
+
+	if tokenInNormFactor.IsZero() {
+		return osmomath.ZeroBigDec(), domain.ZeroNormalizationFactorError{Denom: tokenIn.Denom, PoolId: r.GetId()}
+	}
+
+	if tokenOutNormFactor.IsZero() {
+		return osmomath.ZeroBigDec(), domain.ZeroNormalizationFactorError{Denom: tokenOutDenom, PoolId: r.GetId()}
+	}
```

Comment:
In Go, it is idiomatic to return a nil value rather than an initialized zero when an error is returned.

Would it make sense to return an uninitialized `osmomath.BigDec{}` here for consistency